aten/src/ATen/core/ivalue.h

```diff
@@ -552,12 +560,20 @@ struct C10_EXPORT ivalue::Future final : c10::intrusive_ptr_target {
       value_ = std::move(value);
     }
 
-    // There is no need to protect callbacks anymore.
-    // Once completed_ is set to true, no one can add new callback to the list.
-    for (auto& callback : callbacks) {
-      callback();
+    fireCallbacks();
+  }
+
+  void markCompleted(const std::string& error_msg, bool is_jit_exception) {
```

Comment:
nit: this function should just take a FutureError object. If we change the arguments to its constructor then it prevents having to change the type signature of multiple functions.